bfdriver/src/platform/efi/efi_base/platform.c

```diff
@@ -312,3 +176,33 @@ platform_unload_info(struct platform_info_t *info)
 {
     (void) info;
 }
+
+int printf(const char *format, ...)
```

Comment:
??? What is this for?

Reply:
Upon revisiting this, It looks like bfelf_loader and common can attempt to output error messages with printf, which isn't present in this environment.  I opted to use return values to detect errors instead, but could wrap the EFI function for console output here